### ISSUE TITLE
[ProgressIndicator] setProgressCompat with animation results in memory leak

### ISSUE THREAD

**Issue by ditn** [material-components/material-components-android#1440]
**Description:** When calling `setProgressCompat(progress = 100, animated = true)`, LeakCanary is informing me of a leak of the underlying `Activity`. For sanity's sake, I've checked that `setProgressCompat(progress = 100, animated = false)` doesn't leak.

Full trace here:

```
┬───
│ GC Root: System class
│
├─ android.os.Looper class
│    Leaking: NO (Thread↓ is not leaking and a class is never leaking)
│    ↓ static Looper.sMainLooper
├─ android.os.Looper instance
│    Leaking: NO (Thread↓ is not leaking)
│    ↓ Looper.mThread
├─ java.lang.Thread instance
│    Leaking: NO (the main thread always runs)
│    Thread name: 'main'
│    ↓ Thread.threadLocals
│             ~~~~~~~~~~~~
├─ java.lang.ThreadLocal$ThreadLocalMap instance
│    Leaking: UNKNOWN
│    ↓ ThreadLocal$ThreadLocalMap.table
│                                 ~~~~~
├─ java.lang.ThreadLocal$ThreadLocalMap$Entry[] array
│    Leaking: UNKNOWN
│    ↓ ThreadLocal$ThreadLocalMap$Entry[].[21]
│                                         ~~~~
├─ java.lang.ThreadLocal$ThreadLocalMap$Entry instance
│    Leaking: UNKNOWN
│    ↓ ThreadLocal$ThreadLocalMap$Entry.value
│                                       ~~~~~
├─ android.animation.AnimationHandler instance
│    Leaking: UNKNOWN
│    ↓ AnimationHandler.mAnimationCallbacks
│                       ~~~~~~~~~~~~~~~~~~~
├─ java.util.ArrayList instance
│    Leaking: UNKNOWN
│    ↓ ArrayList.elementData
│                ~~~~~~~~~~~
├─ java.lang.Object[] array
│    Leaking: UNKNOWN
│    ↓ Object[].[1]
│               ~~~
├─ android.animation.AnimatorSet instance
│    Leaking: UNKNOWN
│    ↓ AnimatorSet.mListeners
│                  ~~~~~~~~~~
├─ java.util.ArrayList instance
│    Leaking: UNKNOWN
│    ↓ ArrayList.elementData
│                ~~~~~~~~~~~
├─ java.lang.Object[] array
│    Leaking: UNKNOWN
│    ↓ Object[].[0]
│               ~~~
├─ com.google.android.material.progressindicator.LinearIndeterminateDrawable$4 instance
│    Leaking: UNKNOWN
│    Anonymous subclass of android.animation.AnimatorListenerAdapter
│    ↓ LinearIndeterminateDrawable$4.this$0
│                                    ~~~~~~
├─ com.google.android.material.progressindicator.LinearIndeterminateDrawable instance
│    Leaking: UNKNOWN
│    ↓ LinearIndeterminateDrawable.context
│                                  ~~~~~~~
╰→ co.cuvva.presentation.ltm.about.you.ui.AboutYouActivity instance
​     Leaking: YES (ObjectWatcher was watching this because co.cuvva.presentation.ltm.about.you.ui.AboutYouActivity received Activity#onDestroy() callback and Activity#mDestroyed is true)
​     key = 64ef07bf-c99d-4498-b3e8-d91df1cf5eca
​     watchDurationMillis = 34092
​     retainedDurationMillis = 29091

METADATA

Build.VERSION.SDK_INT: 29
Build.MANUFACTURER: Google
LeakCanary version: 2.4
App process name: co.cuvva.hourly
Analysis duration: 9845 ms
```

**Expected behavior:** No leak.

**Source code:** The code snippet which is causing this issue. Please consider attaching a minimal sample app that reproduces the issue.

**Android API version:** 29

**Material Library version:** 1.3.0-alpha01

**Device:** Pixel 2 Emulator, version 30.0.18

**Comment by pekingme** [comment]
Hi @ditn, sorry for the late reply. I noticed that the leak happens in a LinearIndeterminateDrawable. Could you provide more information about how the leaking Progress Indicator is used in Cuvva? For example, is it used in the indeterminate mode? Does it set progress on such indeterminate PI? If so, are you able to identify if the leak happens right after `setProgressCompat(100, true)` is called, after the indeterminate animation ends or after the progress animation ends?

**Comment by ditn** [comment]
No problem! We set `android:indeterminate="false"` in XML, and having investigated some more it appears that the `ProgressIndicator` leaks when leaving the page - whether you set progress or not doesn't matter. I've created a repo to recreate it reliably [here](https://github.com/ditn/ProgressIndicatorLeak). Apologies for the initial incorrect report.

Steps to reproduce:

* Start app, see `ProgressIndicator`
* Click `Close Activity` button
* Wait a few seconds
* See notification from LeakCanary, "3 retained objects, tap to dump heap"
* Click to dump
* See notication, "Found 1 retained object grouped as 1 distinct leak"
* See leak + heap dump.

Note that editing the XML so that `android:indeterminate="true"` seems to prevent the leak.
